realm/realm-library/src/objectServer/java/io/realm/mongodb/mongo/MongoCollection.java

```diff
@@ -805,4 +814,101 @@ public ResultT run() {
             }
         });
     }
+
+    /**
+     * Watches a collection. The resulting stream will be notified of all events on this collection
+     * that the active user is authorized to see based on the configured MongoDB Realm rules.
+     *
+     * @return a task that provides access to the stream of change events.
+     */
+    public RealmEventStreamTask<DocumentT> watch() {
+        return new RealmEventStreamTaskImpl<>(getNamespace().getFullName(),
+                new RealmEventStreamTaskImpl.Executor<DocumentT>() {
+                    @Override
+                    public EventStream<DocumentT> run() throws IOException {
+                        return osMongoCollection.watch();
+                    }
+                });
+    }
+
+    /**
+     * Watches specified IDs in a collection.
+     *
+     * @param ids the ids to watch.
+     * @return a task that provides access to the stream of change events.
+     */
+    public RealmEventStreamTask<DocumentT> watch(final BsonValue... ids) {
+        return new RealmEventStreamTaskImpl<>(getNamespace().getFullName(),
+                new RealmEventStreamTaskImpl.Executor<DocumentT>() {
+                    @Override
+                    public EventStream<DocumentT> run() throws IOException {
+                        return osMongoCollection.watch(ids);
+                    }
+                });
+    }
+
+    /**
+     * Watches specified IDs in a collection. This convenience overload supports the use case
+     * of non-{@link BsonValue} instances of {@link ObjectId} by wrapping them in
+     * {@link BsonObjectId} instances for the user.
+     *
+     * @param ids unique object identifiers of the IDs to watch.
+     * @return a task that provides access to the stream of change events.
+     */
+    public RealmEventStreamTask<DocumentT> watch(final ObjectId... ids) {
+        return new RealmEventStreamTaskImpl<>(getNamespace().getFullName(),
+                new RealmEventStreamTaskImpl.Executor<DocumentT>() {
+                    @Override
+                    public EventStream<DocumentT> run() throws IOException {
+                        return osMongoCollection.watch(ids);
+                    }
+                });
+    }
+
+    /**
+     * Watches a collection. The provided document will be used as a match expression filter on
+     * the change events coming from the stream. This convenience overload supports the use of
+     * non-{@link BsonDocument} instances for the user.
+     * <p>
+     * See https://docs.mongodb.com/manual/reference/operator/aggregation/match/ for documentation
```

Comment:
These should probably be wrapped in `<a href ...` tags to render correctly?

Reply:
Yes, it needs the <a> tag!